cot/src/form/types.rs

```diff
@@ -0,0 +1,383 @@
+//! Form Field Types for Cot
+//!
+//! This module provides a collection of form field types and utilities for
+//! validating, parsing, and converting user input within Cot. It includes
+//! general-purpose newtype wrappers and associated trait implementations to
+//! ensure consistent and safe processing of form data.
+
+use std::fmt::Debug;
+use std::str::FromStr;
+
+#[cfg(feature = "db")]
+use cot::db::impl_mysql::MySqlValueRef;
+#[cfg(feature = "db")]
+use cot::db::impl_postgres::PostgresValueRef;
+#[cfg(feature = "db")]
+use cot::db::impl_sqlite::SqliteValueRef;
+use email_address::EmailAddress;
```

Comment:
Each of these `impl` modules is feature flag-gated behind a separate feature flag. You should therefore use the following instead:

```suggestion
#[cfg(feature = "mysql")]
use cot::db::impl_mysql::MySqlValueRef;
#[cfg(feature = "postgres")]
use cot::db::impl_postgres::PostgresValueRef;
#[cfg(feature = "sqlite")]
use cot::db::impl_sqlite::SqliteValueRef;
use email_address::EmailAddress;
```